### ISSUE TITLE
"warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release" on Bazel 0.19.0rc6 and master

### ISSUE THREAD

**Issue by davido** [bazelbuild/bazel#6433]
On Bazel 0.19.0rc6 I'm seeing this warning during build of gerrit and gerrit plugins:

```
  $ bazel build reviewers
INFO: Build options have changed, discarding analysis cache.
INFO: Analysed target //:reviewers (36 packages loaded, 527 targets configured).
INFO: Found 1 target...
INFO: From SkylarkAction external/bazel_tools/tools/jdk/platformclasspath.jar [for host]:
warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release
INFO: From GWT compiling reviewers__gwt_application.zip [for host]:
Compiling module com.googlesource.gerrit.plugins.reviewers.ReviewersForm
[...]
```
I'm seeing the same warning on Bazel master as well (d48e0e27600deeed9eab01d6e42ab4b6e715d7a6). This seems to be a regression, as all is fine on Bazel 0.18.0:

```
  $ bazel version
Build label: 0.18.0

  $ bazel build reviewers
Starting local Bazel server and connecting to it...
INFO: Analysed target //:reviewers (36 packages loaded).
INFO: Found 1 target...
INFO: From GWT compiling reviewers__gwt_application.zip [for host]:
Compiling module com.googlesource.gerrit.plugins.reviewers.ReviewersForm
   Compiling 5 permutations
      Compiling permutation 0...
      Compiling permutation 1...
      Compiling permutation 2...
      Compiling permutation 3...
      Compiling permutation 4...
   Compile of permutations succeeded
   Compilation succeeded -- 24.064s
Linking into /home/davido/.cache/bazel/_bazel_davido/6bfabb416f8d36020b7be6ab9749326b/sandbox/linux-sandbox/7/execroot/reviewers/bazel-out/host/bin/reviewers__gwt_application.zip.gwt_output/reviewers
   Link succeeded
   Linking succeeded -- 0.188s
Target //:reviewers up-to-date:
  bazel-genfiles/reviewers.jar
INFO: Elapsed time: 44.457s, Critical Path: 33.80s
INFO: 13 processes: 10 linux-sandbox, 3 worker.
INFO: Build completed successfully, 15 total actions
``` 

//CC @cushon.

**Comment by cushon** [comment]
What combinations of Java toolchains and JDKs is that build using?

**Comment by davido** [comment]
Only java 8:


```
  $JAVA_HOME/bin/java -version
openjdk version "1.8.0_171"
OpenJDK Runtime Environment (IcedTea 3.8.0) (build 1.8.0_171-b11 suse-1.3-x86_64)
OpenJDK 64-Bit Server VM (build 25.171-b11, mixed mode)
```

**Comment by cushon** [comment]
The message was introduced by 999354b81a0639700cb460e9e49af858af1cbaca.

It's supposed to warn about builds that are using a JDK 9 `--javabase` but targeting the Java 8 language level. In that situation we use the infrastructure added to support the javac `--release` to get a bootclasspath for cross-compilation, but that approach doesn't provide access to internal (e.g. `sun.misc.*`) APIs, see: #6179.

Example:

```
./bazel-0.19.0rc6-linux-x86_64 build --define=ABSOLUTE_JAVABASE=$JAVA9_HOME \
  --javabase @bazel_tools//tools/jdk:absolute_javabase ...
...
warning: could not find a JDK 8 bootclasspath in $JAVA9_HOME, falling back to --release
```

I'm surprised you're seeing that warning with a JDK 8 `--javabase`. That shouldn't happen. Do you have a repo? Does `$JAVA_HOME/jre/lib/rt.jar` exist on your system?

**Comment by mhlopko** [comment]
cc @iirina

**Comment by katre** [comment]
Potential release loker for #6116.

**Comment by davido** [comment]
> Does $JAVA_HOME/jre/lib/rt.jar exist on your system?

Yes, it does exist:

```
  $ ls -all $JAVA_HOME/jre/lib/rt.jar
-rw-r--r-- 1 root root 34730768 Jul 13 16:53 /usr/lib64/jvm/java/jre/lib/rt.jar
```

> Do you have a repo?

I can consistently reproduce it on gerrit reviewers plugin:

```
  $ git clone https://gerrit.googlesource.com/plugins/reviewers
Cloning into 'reviewers'...
remote: Total 2437 (delta 555), reused 2437 (delta 555)
Receiving objects: 100% (2437/2437), 474.39 KiB | 4.61 MiB/s, done.
Resolving deltas: 100% (555/555), done.

  $ cd reviewers

  $ bazel version
Build label: 0.19.0rc6

  $ bazel build reviewers
Starting local Bazel server and connecting to it...
INFO: Repository rule 'com_googlesource_gerrit_bazlets' returned: {"remote": "https://gerrit.googlesource.com/bazlets", "commit": "3a53199198db3f49a43b7708c2c3352670393717", "shallow_since": "2018-06-08", "init_submodules": False, "verbose": False, "strip_prefix": "", "patches": [], "patch_tool": "patch", "patch_args": ["-p0"], "patch_cmds": [], "name": "com_googlesource_gerrit_bazlets"}
INFO: Analysed target //:reviewers (36 packages loaded, 530 targets configured).
INFO: Found 1 target...
INFO: From SkylarkAction external/bazel_tools/tools/jdk/platformclasspath.jar [for host]:
warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release
INFO: From GWT compiling reviewers__gwt_application.zip [for host]:
Compiling module com.googlesource.gerrit.plugins.reviewers.ReviewersForm
   Compiling 5 permutations
      Compiling permutation 0...
      Compiling permutation 1...
      Compiling permutation 2...
      Compiling permutation 3...
      Compiling permutation 4...
   Compile of permutations succeeded
   Compilation succeeded -- 26.547s
Linking into /home/davido/.cache/bazel/_bazel_davido/bd29d79aeb6cdca7a3276c675474c0e8/sandbox/linux-sandbox/11/execroot/reviewers/bazel-out/host/bin/reviewers__gwt_application.zip.gwt_output/reviewers
   Link succeeded
   Linking succeeded -- 0.197s
Target //:reviewers up-to-date:
  bazel-genfiles/reviewers.jar
INFO: Elapsed time: 52.376s, Critical Path: 38.49s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 18 processes: 15 linux-sandbox, 3 worker.
INFO: Build completed successfully, 23 total actions
```

**Comment by davido** [comment]
And I also see this on gerrit itself:

```
  $ cd gerrit
  $ tools/eclipse/project.py 
INFO: Analysed target //tools/eclipse:main_classpath_collect (297 packages loaded, 4656 targets configured).
INFO: Found 1 target...
Target //tools/eclipse:main_classpath_collect up-to-date:
  bazel-bin/tools/eclipse/main_classpath_collect.runtime_classpath
INFO: Elapsed time: 6.350s, Critical Path: 0.15s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 0 processes.
INFO: Build completed successfully, 2 total actions
INFO: Analysed target //tools/eclipse:gwt_classpath_collect (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //tools/eclipse:gwt_classpath_collect up-to-date:
  bazel-bin/tools/eclipse/gwt_classpath_collect.runtime_classpath
INFO: Elapsed time: 0.330s, Critical Path: 0.16s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 0 processes.
INFO: Build completed successfully, 2 total actions
INFO: Analysed target //tools/eclipse:autovalue_classpath_collect (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //tools/eclipse:autovalue_classpath_collect up-to-date:
  bazel-bin/tools/eclipse/autovalue_classpath_collect.runtime_classpath
INFO: Elapsed time: 0.323s, Critical Path: 0.17s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 0 processes.
INFO: Build completed successfully, 2 total actions
INFO: Analysed 3 targets (0 packages loaded, 1 target configured).
INFO: Found 3 targets...
INFO: From SkylarkAction external/bazel_tools/tools/jdk/platformclasspath.jar [for host]:
warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release
[...]
```

**Comment by cushon** [comment]
> I can consistently reproduce it on gerrit reviewers plugin:

When I try to build that it tries to find `gerrit-plugin-api-2.16-SNAPSHOT.jar` in my local maven repository and then fails. Are there docs for how to get set up to reproduce?

**Comment by davido** [comment]
Ah, right, you would need to publish gerrit plugin API, first.

Clone gerrit and run:

```
  $ VERBOSE=1 tools/maven/api.sh install
+ bazel build //tools/maven:gen_api_install
INFO: Analysed target //tools/maven:gen_api_install (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //tools/maven:gen_api_install up-to-date:
  bazel-genfiles/tools/maven/api_install.sh
INFO: Elapsed time: 0.362s, Critical Path: 0.21s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
+ ./bazel-genfiles/tools/maven/api_install.sh
+ bazel build //java/com/google/gerrit/acceptance:framework_deploy.jar //java/com/google/gerrit/extensions:extension-api_deploy.jar //plugins:plugin-api_deploy.jar //gerrit-plugin-gwtui:gwtui-api_deploy.jar //java/com/google/gerrit/acceptance:libframework-lib-src.jar //java/com/google/gerrit/extensions:libapi-src.jar //plugins:plugin-api-sources_deploy.jar //gerrit-plugin-gwtui:gwtui-api-source_deploy.jar //java/com/google/gerrit/acceptance:framework-javadoc //java/com/google/gerrit/extensions:extension-api-javadoc //plugins:plugin-api-javadoc //gerrit-plugin-gwtui:gwtui-api-javadoc
INFO: Analysed 12 targets (0 packages loaded, 0 targets configured).
INFO: Found 12 targets...
INFO: Elapsed time: 0.348s, Critical Path: 0.19s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
+ python tools/maven/mvn.py -v 3.0-SNAPSHOT -s gerrit-acceptance-framework:jar:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/framework_deploy.jar -s gerrit-extension-api:jar:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/extension-api_deploy.jar -s gerrit-plugin-api:jar:bazel-out/k8-fastbuild/bin/plugins/plugin-api_deploy.jar -s gerrit-plugin-gwtui:jar:bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api_deploy.jar -s gerrit-acceptance-framework:java-source:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/libframework-lib-src.jar -s gerrit-extension-api:java-source:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/libapi-src.jar -s gerrit-plugin-api:java-source:bazel-out/k8-fastbuild/bin/plugins/plugin-api-sources_deploy.jar -s gerrit-plugin-gwtui:java-source:bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api-source_deploy.jar -s gerrit-acceptance-framework:javadoc:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/framework-javadoc.zip -s gerrit-extension-api:javadoc:bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/extension-api-javadoc.zip -s gerrit-plugin-api:javadoc:bazel-out/k8-fastbuild/bin/plugins/plugin-api-javadoc.zip -s gerrit-plugin-gwtui:javadoc:bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api-javadoc.zip -a install
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-acceptance-framework_pom.xml -Dpackaging=jar -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/framework_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-extension-api_pom.xml -Dpackaging=jar -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/extension-api_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-api_pom.xml -Dpackaging=jar -Dfile=bazel-out/k8-fastbuild/bin/plugins/plugin-api_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-gwtui_pom.xml -Dpackaging=jar -Dfile=bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-acceptance-framework_pom.xml -Dpackaging=java-source -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/libframework-lib-src.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-extension-api_pom.xml -Dpackaging=java-source -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/libapi-src.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-api_pom.xml -Dpackaging=java-source -Dfile=bazel-out/k8-fastbuild/bin/plugins/plugin-api-sources_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-gwtui_pom.xml -Dpackaging=java-source -Dfile=bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api-source_deploy.jar
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-acceptance-framework_pom.xml -Dpackaging=javadoc -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/acceptance/framework-javadoc.zip
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-extension-api_pom.xml -Dpackaging=javadoc -Dfile=bazel-out/k8-fastbuild/bin/java/com/google/gerrit/extensions/extension-api-javadoc.zip
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-api_pom.xml -Dpackaging=javadoc -Dfile=bazel-out/k8-fastbuild/bin/plugins/plugin-api-javadoc.zip
mvn install:install-file -Dversion=3.0-SNAPSHOT -DpomFile=/home/davido/projects/gerrit2/tools/maven/gerrit-plugin-gwtui_pom.xml -Dpackaging=javadoc -Dfile=bazel-out/k8-fastbuild/bin/gerrit-plugin-gwtui/gwtui-api-javadoc.zip
Version: 3.0-SNAPSHOT
```

However, recently, Gerrit snapshot was bumped to 3.0-SNAPSHOT, so that you would need to update reviewers plugin, to build against that snapshot version, by bumping bazlets version in WORKSPACE in reveiwers plugin:

```
  $ diff --git a/WORKSPACE b/WORKSPACE
index 0d61018..c697b86 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -3,7 +3,7 @@ workspace(name = "reviewers")
 load("//:bazlets.bzl", "load_bazlets")
 
 load_bazlets(
-    commit = "3a53199198db3f49a43b7708c2c3352670393717",
+    commit = "e4662688b3f482cf5789b27a1ea726e8d0aabfc8",
     #    local_path = "/home/<user>/projects/bazlets",
 )
```

That was done earlier today: https://gerrit-review.googlesource.com/c/bazlets/+/200630.

Reviewers plugin wasn't updated yet on master.

Now you should be able to say:

```
  $ bazel build reviewers
Starting local Bazel server and connecting to it...
INFO: Repository rule 'com_googlesource_gerrit_bazlets' returned: {"remote": "https://gerrit.googlesource.com/bazlets", "commit": "e4662688b3f482cf5789b27a1ea726e8d0aabfc8", "shallow_since": "2018-10-18", "init_submodules": False, "verbose": False, "strip_prefix": "", "patches": [], "patch_tool": "patch", "patch_args": ["-p0"], "patch_cmds": [], "name": "com_googlesource_gerrit_bazlets"}
INFO: Analysed target //:reviewers (36 packages loaded, 530 targets configured).
INFO: Found 1 target...
INFO: From SkylarkAction external/bazel_tools/tools/jdk/platformclasspath.jar [for host]:
warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release
[...]
```

**Comment by davido** [comment]
A simpler option to reproduce would be to use this change: [1], that updated the API to 2.16-rc0, that was released today morning to Maven Central: [2]. With this change: [1] cherry-picked on top of master of reviewers plugin, you could just build, without publishing snapshot version.

* [1] https://gerrit-review.googlesource.com/c/plugins/reviewers/+/200671 https://groups.google.com/d/topic/repo-discuss/7ufIRFCaKxU/discussion
* [2] https://groups.google.com/d/topic/repo-discuss/7ufIRFCaKxU/discussion

**Comment by cushon** [comment]
Thanks, I was able to repo with those instructions.

This is an issue with host-targeting-host compilations, since the default host JDK is 9 but we're still using the Java 8 language level in the host, and we can't find a JDK 8 bootclasspath in the host JDK.

```
java_binary(
    name = "b",
    srcs = ["B.java"],
    main_class = "B",
)
```

```
genrule(
    name = "g",
    outs = ["o"],
    cmd = "$(location b) > $@",
    tools = ["b"],
)
public class B {
  public static void main(String[] args) {
    System.err.println("hello");
  }
}
```

```
./bazel-0.19.0rc6-linux-x86_64 build :g
INFO: Analysed target //:g (15 packages loaded, 230 targets configured).
INFO: Found 1 target...
INFO: From SkylarkAction external/bazel_tools/tools/jdk/platformclasspath.jar [for host]:
warning: could not find a JDK 8 bootclasspath in external/embedded_jdk, falling back to --release
```

**Comment by cushon** [comment]
Here's a work-around that makes the current behaviour explicit and fixes the warning spam:

```diff
diff --git a/src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java b/src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java
index b56e784c00..e8dfe53968 100644
--- a/src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java
+++ b/src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java
@@ -97,7 +97,7 @@ public class JavaOptions extends FragmentOptions {
 
   @Option(
     name = "host_java_toolchain",
-    defaultValue = "@bazel_tools//tools/jdk:toolchain",
+    defaultValue = "@bazel_tools//tools/jdk:toolchain_host",
     converter = LabelConverter.class,
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
diff --git a/tools/jdk/BUILD b/tools/jdk/BUILD
index 53b72d65c1..80f33d4dc4 100644
--- a/tools/jdk/BUILD
+++ b/tools/jdk/BUILD
@@ -202,6 +202,18 @@ bootclasspath(
     target_javabase = "current_java_runtime",
 )
 
+# For use with host-targeting-host builds.
+#
+# The default host JDK is > 8 but we're using the Java 8 language level on the
+# host, so we have no choice but to use --release for host-targeting-host
+# builds.
+bootclasspath(
+    name = "platformclasspath8",
+    src = "DumpPlatformClassPath.java",
+    host_javabase = "current_java_runtime",
+    release = "8",
+)
+
 default_java_toolchain(
     name = "toolchain_hostjdk8",
     bootclasspath = [":platformclasspath"],
@@ -220,6 +232,15 @@ default_java_toolchain(
     target_version = "8",
 )
 
+# The default --host_java_toolchain
+default_java_toolchain(
+    name = "toolchain_host",
+    bootclasspath = [":platformclasspath8"],
+    jvm_opts = JDK9_JVM_OPTS,
+    source_version = "8",
+    target_version = "8",
+)
+
 # The 'vanilla' toolchain is an unsupported alternative to the default.
 #
 # It does not provider any of the following features:
```

I don't have an opinion about whether this should be a release blocker. It isn't a functional regression, it's 'just' log spam.

@iirina and @lberki, WDYT?

**Comment by davido** [comment]
@cushon Thanks for the quick fix. I confirmed, when applying that patch on Bazel master and building 
reviewers plugin, the warning is gone.

Can you apply this patch on Bazel master? If Bazel teams decided not to fix this log spam regression in upcoming 0.19 release, we could skip that release and wait for the next release where this problem is fixed.

**Comment by katre** [comment]
If the fix is this simple, I'd prefer to eliminate the confusing logs and cherrypick this into 0.19.0.

@iirina, @lberki can you comment?

**Comment by lberki** [comment]
Would be nice, but don't we only cherrypick for regressions?

**Comment by katre** [comment]
Oh, I thought the log message was a regression. Is it also present in 0.18.0?

**Comment by davido** [comment]
> Is it also present in 0.18.0?

This error message is not present in 0.18.0.

**Comment by katre** [comment]
Given the lack of comments either way, and that there's no actual commit for me to cherrypick at this point, I'm unmarking this as a release blocker.

Please do consider submitting a commit to fix the log spam so it can make the next release.

**Comment by cushon** [comment]
To clarify, the warning didn't happen in earlier versions, so that part was a change post-0.18. The warning doesn't result in broken/incorrect builds. I don't know how regressions are defined for the purposes of release blockers. I'm still happy to defer the decision, I just want to make sure everyone has the same information.

The fix is out for review and would be an easy cherry-pick if you want to go that route.

**Comment by lberki** [comment]
Fair enough -- I wouldn't cherry-pick the fix because I prefer some log spam to a chance of breakage, but I don't have a very strong opinion, so if you want to do it, go ahead.

**Comment by katre** [comment]
Is there a commit yet that addresses this?

**Comment by cushon** [comment]
@katre cl/217820713, but I haven't been able to submit it yet because the presubmit is down.

**Comment by katre** [comment]
@cushon You should be able to sync to head and retry today.

**Comment by cushon** [comment]
and now the CI is broken :/ https://buildkite.com/bazel/bazel-bazel/builds/5122

**Comment by katre** [comment]
Sending a quick change to disable the failing test, will let you know when it's in

**Comment by katre** [comment]
It's in.

**Comment by cushon** [comment]
The fix didn't stick:

```
src/main/java/com/google/devtools/build/lib/unsafe/UnsafeProvider.java:21: error: symbol not found sun.misc.Unsafe
import sun.misc.Unsafe;
```

https://storage.googleapis.com/bazel-buildkite-artifacts/ad153be8-d848-4ccc-b0df-4f0c631b1ab3/src/test/shell/bazel/bazel_determinism_test/attempt_1.log

The issue is that using the host JDK 9's `--release` support instead of its default bootclasspath regresses #6179; `--release 8` on JDK 9 doesn't include e.g. `sun.misc.Unsafe`.

I think the current behaviour is actually what we want. host-targeting-host compilations with a JDK 9 host_javabase are actually targeting JDK 9, so it's OK to use a JDK 9 bootclasspath. It's also OK to limit ourselves to the Java 8 language level.

Maybe we should just disable that warning, either for this specific use of DumpPlatformClasspath or in general.

**Comment by ittaiz** [comment]
Just to make sure I understand- using host_javabase and javabase or Oracle
jdk 8 will work fine with sun.misc.Unsafe right?

**Comment by cushon** [comment]
> using host_javabase and javabase or Oracle jdk 8 will work fine with sun.misc.Unsafe right?

Yes: `sun.misc.Unsafe` should be available in all supported configurations, thanks to the fix for #6179.

The warning tracked by this bug is unfortunate, but the functionality hasn't regressed.

**Comment by cushon** [comment]
I submitted a different fix: 917d0cd725aa6c796573dbafd3cb78ec484ee40a.
